src/components/Layout/Nav.jsx

```diff
@@ -34,38 +50,49 @@ const Nav = () => {
                             id="name"
                             type="text"
                             className="input-field"
-                            autoComplete="off"
+                            autocomplete="off"
                             placeholder=" "
-
+                            value={searchTxt}
+                            onChange={(e) => setSearchTxt(e.target.value)}
+                            onKeyDown={searchHandler}
 
                         />
-                        <span className="iconify icons text-2">
+                        <span onClick={() => {
+                            dispatch({ type: "SEARCH_PRODUCT", payload: searchTxt });
+                            setSearchTxt("");
+                        }} className="iconify icons text-2">
                             <Icon icon="ant-design:search-outlined" />
                         </span>
 
-                        <label htmlFor="name" className="input-label text-2">
+                        <label for="name" className="input-label text-2">
```

Comment:
in react's JSX htmlFor is used for "for" attribute. Other than that everything looks great. Go ahead and merge the branch.